rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderImplTests.java

```diff
@@ -686,6 +696,65 @@ void getOldFederationBtcUTXOs_calledTwice_returnCachedUtxos() {
 
     }
 
+    @Test
+    void getFederationElection_whenElectionIsInStorage_shouldReturnNewElection() {
+
+        // Arrange
+
+        AddressBasedAuthorizer authorizer = getTestingAddressBasedAuthorizer();
+
+        ABICallElection expectedElection = getSampleElection("function1", authorizer);
+        byte[] expectedElectionEncoded = BridgeSerializationUtils.serializeElection(expectedElection);
+
+        StorageAccessor storageAccessor = new InMemoryStorage();
+        storageAccessor.saveToRepository(FEDERATION_ELECTION_KEY.getKey(), expectedElectionEncoded);
+
+        FederationStorageProvider federationStorageProvider = new FederationStorageProviderImpl(storageAccessor);
+
+        // Act
+
+        ABICallElection actualElection = federationStorageProvider.getFederationElection(authorizer);
+
+        // Assert
+
+        assertArrayEquals(expectedElectionEncoded, serializeElection(actualElection));
+
+        // Setting a different election in storage to assert that calling the method again returns the cached election
+
+        ABICallElection secondElectionSample = getSampleElection("function2", authorizer);
+        storageAccessor.saveToRepository(FEDERATION_ELECTION_KEY.getKey(), BridgeSerializationUtils.serializeElection(secondElectionSample));
+
+        ABICallElection cachedElection = federationStorageProvider.getFederationElection(authorizer);
+
+        assertArrayEquals(serializeElection(actualElection), serializeElection(cachedElection));
+        assertFalse(Arrays.equals(expectedElectionEncoded, serializeElection(secondElectionSample)));
+
+    }
+
+    @Test
+    void getFederationElection_whenElectionIsNotInStorage_shouldReturnDefault() {
+
+        // Arrange
+
+        AddressBasedAuthorizer authorizer = getTestingAddressBasedAuthorizer();
+
+        ABICallElection expectedElection = new ABICallElection(authorizer);
+        byte[] expectedElectionEncoded = BridgeSerializationUtils.serializeElection(expectedElection);
```

Comment:
Let's move this variable to the `assert` section. I think it's better to have the `expected` variables close to the assertion where is being used(when possible, of course). I think this case it's suitable for being there.